sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -188,45 +174,44 @@ private static void WriteMemory(MemoryStream stream, ReadOnlyMemory<byte> memory
         /// ]]></code>
         /// </example>
         /// <remarks> See error response code and error response message for more detail. </remarks>
-        public virtual Response<UploadLogsResult> Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
+        public virtual Response Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(ruleId, nameof(ruleId));
             Argument.AssertNotNullOrEmpty(streamName, nameof(streamName));
             Argument.AssertNotNullOrEmpty(logs, nameof(logs));
 
             using var scope = ClientDiagnostics.CreateScope("LogsIngestionClient.Upload");
 
-            RequestContext requestContext = GenerateRequestContext(cancellationToken);
             Response response = null;
-            List<UploadLogsError> errors = new List<UploadLogsError>();
+            var exceptions = new List<Exception>();
+
             scope.Start();
-            try
+            // Partition the stream into individual blocks
+            foreach (BatchedLogs<T> batch in Batch(logs, options)) //TODO: check if we should throw early for sync case? as soon as we find exception
```

Comment:
Do we need these TODOs or have they been figured out?

Reply:
> Maybe we could make the partial failure clearer in the `AggregateException`'s `Message`? Imagine something like `$"{failedCount} of {totalCount} log entries failed to upload. See the InnerExceptions for more details."`. We could also add entries to the exception's `Data` in the future if we find customers have any scenarios to access that programmatically.

It might be nice if we could somehow also include the indices of the failed log entries from the initial list, e.g. "Upload failed for indices 4,5,6,12,13,14,15"